llvm/test/CodeGen/AArch64/sve-float.ll

```diff
@@ -0,0 +1,71 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 3
+; RUN: llc < %s -mtriple aarch64 -mattr=+sve  | FileCheck %s
+; RUN: llc < %s -mtriple aarch64 -mattr=+sve -global-isel -aarch64-enable-gisel-sve=1 | FileCheck %s
+
+;; fneg
+define <vscale x 2 x double> @fnegnxv2double(<vscale x 2 x double> %a) {
+; CHECK-LABEL: fnegnxv2double:
+; CHECK:       // %bb.0: // %entry
+; CHECK-NEXT:    ptrue p0.d
+; CHECK-NEXT:    fneg z0.d, p0/m, z0.d
+; CHECK-NEXT:    ret
+entry:
+  %c = fneg <vscale x 2 x double> %a
+  ret <vscale x 2 x double> %c
+}
+
```

Comment:
Test 3 x case?

Reply:
Assertion failed: ((TypeSize::ScalarTy)SrcOps.size() * SrcOps[0].getLLTTy(*getMRI()).getSizeInBits() == DstOps[0].getLLTTy(*getMRI()).getSizeInBits() && "input vectors do not exactly cover the output vector register"), function buildInstr, file MachineIRBuilder.cpp, line 1419.

Reply:
No! There are no illegal ***vscale*** types.

Reply:
? It's just more bugs

Reply:
For AArch64 SVE, the size of the granule is 128 bit. It is your choice how to fill it:
* 2 x 64bit
* 4 x 32bit
* 8 x 16bit
We cannot scalarize or widen scalable vectors. The **vscale** is unknown  at compile time.

Reply:
Can always loop. In any case, assert in the compiler is not an OK failure mode

Reply:
The DAG error message could be improved. Tell me there were scalable vectors. The GlobalISel assert is not helpful at all.

Reply:
> We cannot scalarize or widen scalable vectors. The vscale is unknown at compile time.

We can widen scalable vector arithmetic just fine. It's loads/store that are broken because we can't access more memory. It would be kind of messy, but it should be possible to generate a mask to block off the extra elements so a scalable load/store could be widened. Just need a step_vector, a splat of (vscale * original_min_elts), and a compare I think.

Reply:
G_STEP_VECTOR is still missing. Can this be target independent?